### ISSUE TITLE
Improve the efficiency of the RMSNorm aggregation 

### ISSUE THREAD

**Issue by lancerts** [linkedin/Liger-Kernel#179]
### 🚀 The feature, motivation and pitch

Move this line https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/rms_norm.py#L306, the sum in pytorch to partial aggregation in triton, reference 
https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/layer_norm.py#L106,
which does 2 level of aggregation, first in triton and second in torch (more efficient).

**Comment by ByronHsu** [comment]
cc @AndreSlavescu

**Comment by S1ro1** [comment]
i would like to try this one, #take @lancerts

**Comment by ByronHsu** [comment]
you can refer to layernorm in triton tutorial

**Comment by lancerts** [comment]
The https://triton-lang.org/main/getting-started/tutorials/05-layer-norm.html#sphx-glr-getting-started-tutorials-05-layer-norm-py uses the atomic operation. We can just use similar method in https://github.com/linkedin/Liger-Kernel/blob/main/src/liger_kernel/ops/layer_norm.py#L106, which does not use atomic operation.

**Comment by S1ro1** [comment]
yes, i did go similar to liger layer_norm implementation, however i'm having some issues with numerical stability, is it ok to increase the absolute tolerance in tests? I'll create a draft PR asap, need to fix some issues @lancerts
